File: src/main/java/com/fwai/turtle/service/interfaces/CompanyService.java
package com.fwai.turtle.service.interfaces;

import com.fwai.turtle.dto.CompanyDTO;
import com.fwai.turtle.dto.BankAccountDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;
import com.fwai.turtle.types.CompanyType;

/**
 * Company Service Interface
 * 公司服务接口
 */
public interface CompanyService {
    /**
     * Get paginated list of companies with optional filtering
     * 获取公司列表（分页），支持搜索和状态过滤
     */
    Page<CompanyDTO> getCompanies(Pageable pageable, String search, Boolean active);

    /**
     * Search companies by name
     * 按名称搜索公司
     */
    List<CompanyDTO> searchCompanies(String query);

    /**
     * Get company by ID
     * 根据ID获取公司信息
     */
    CompanyDTO getCompanyById(Long id);

    /**
     * Create new company
     * 创建新公司
     */
    CompanyDTO createCompany(CompanyDTO companyDTO);

    /**
     * Update existing company
     * 更新公司信息
     */
    CompanyDTO updateCompany(Long id, CompanyDTO companyDTO);

    /**
     * Delete company
     * 删除公司
     */
    void deleteCompany(Long id);

    /**
     * Toggle company active status
     * 切换公司启用状态
     */
    CompanyDTO toggleCompanyStatus(Long id);

    /**
     * Get all bank accounts for a company
     * 获取公司的所有银行账户
     */
    List<BankAccountDTO> getCompanyBankAccounts(Long id);

    /**
     * Add a bank account to a company
     * 为公司添加银行账户
     */
    CompanyDTO addBankAccount(Long id, BankAccountDTO bankAccountDTO);

    /**
     * Remove a bank account from a company
     * 移除公司的银行账户
     */
    void removeBankAccount(Long companyId, Long bankAccountId);

    /**
     * Get all active companies
     * 获取所有启用的公司
     */
    List<CompanyDTO> getAllActive();

    /**
     * Find company by type
     * 根据类型查找公司
     */
    CompanyDTO findCompanyByType(CompanyType type);

    /**
     * Set company as primary
     * 设置公司为主公司
     */
    CompanyDTO setPrimary(Long id);
}

File: turtle-web/src/app/services/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Company } from '../models/company.model';
import { environment } from '../../environments/environment';
import { Page } from '../models/page.model';
import { ApiResponse } from '../models/api.model';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {
  private apiUrl = `${environment.apiUrl}/companies`;

  constructor(private http: HttpClient) {}

  getCompanies(params: { page: number; size: number; search?: string; active?: boolean }): Observable<ApiResponse<Page<Company>>> {
    let httpParams = new HttpParams()
      .set('page', params.page.toString())
      .set('size', params.size.toString());

    if (params.search) {
      httpParams = httpParams.set('search', params.search);
    }
    if (params.active !== undefined) {
      httpParams = httpParams.set('active', params.active.toString());
    }

    return this.http.get<ApiResponse<Page<Company>>>(this.apiUrl, { params: httpParams });
  }

  getCompany(id: number): Observable<ApiResponse<Company>> {
    return this.http.get<ApiResponse<Company>>(`${this.apiUrl}/${id}`);
  }

  createCompany(company: Company): Observable<ApiResponse<Company>> {
    return this.http.post<ApiResponse<Company>>(this.apiUrl, company);
  }

  updateCompany(id: number, company: Company): Observable<ApiResponse<Company>> {
    return this.http.put<ApiResponse<Company>>(`${this.apiUrl}/${id}`, company);
  }

  deleteCompany(id: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.apiUrl}/${id}`);
  }

  toggleStatus(id: number): Observable<ApiResponse<Company>> {
    return this.http.put<ApiResponse<Company>>(`${this.apiUrl}/${id}/toggle-status`, {});
  }

  setPrimary(id: number): Observable<ApiResponse<Company>> {
    return this.http.put<ApiResponse<Company>>(`${this.apiUrl}/${id}/set-primary`, {});
  }

  getAllActive(): Observable<ApiResponse<Company[]>> {
    return this.http.get<ApiResponse<Company[]>>(`${this.apiUrl}/active`);
  }

  searchCompanies(query: string): Observable<ApiResponse<Company[]>> {
    const params = new HttpParams().set('query', query);
    return this.http.get<ApiResponse<Company[]>>(`${this.apiUrl}/search`, { params });
  }

  setPrimaryCompany(id: number): Observable<Company> {
    return this.http.put<Company>(`${this.apiUrl}/${id}/set-primary`, {});
  }
}
